python/moltres_xs.py

```diff
@@ -0,0 +1,269 @@
+            self.index = ind
+            self.multi_line_flag = multi
+            self.xs_entry = entry
+
+    def fix_xs(self):
+        for i in range(self.num_burn):
+            for j in range(self.num_uni):
+                for k in range(self.num_temps):
+                    for ii in range(self.num_groups):
+                        start = ii*self.num_groups
+                        stop = start + self.num_groups
+                        self.xs_lib[i][j][k]["REMXS"][ii] += np.sum(
+                            self.xs_lib[i][j][k]["GTRANSFXS"][start:stop])\
+                            - self.xs_lib[i][j][k]["GTRANSFXS"][start+ii]
+                        if self.xs_lib[i][j][k]["FISSE"][ii] != 0:
+                            self.xs_lib[i][j][k]["FISSE"][ii] = (
+                                self.xs_lib[i][j][k]["FISSE"][ii] /
+                                self.xs_lib[i][j][k]["FISSXS"][ii]
+                                )
+
+    def get_xs(self):
+        uni = []
+        L = 0
+        m = 0
+        n = 0
+        lam_temp = 0
+        beta_temp = 0
+        for k, line in enumerate(self.lines):
+            if 'Identifier' in line:
+                val = int(self.lines[k+1].split()[0])
+                if val not in uni:
+                    uni.extend([val])
+                m = uni.index(val)
+                self.xs_lib[L][m][n]['BETA_EFF']\
+                    .extend(beta_temp)
+                self.xs_lib[L][m][n]['DECAY_CONSTANT']\
+                    .extend(lam_temp)
+            if 'branch no.' in line:
+                index = line.find(',')
+                L = int(line[index-4:index])
+                n = int(line.split()[-1])
+            for key in self.catch.keys():
+                if key in line:
+                    if self.catch[key].multi_line_flag:
+                        if (key == 'Betas'):
+                            beta_temp = self.get_multi_line_values(k)
+                        elif(key == 'Lambdas'):
+                            lam_temp = self.get_multi_line_values(k)
+                        else:
+                            self.xs_lib[L][m][n][self.catch[key].xs_entry[0]]\
+                                .extend(
+                                    self.get_multi_line_values(k)
+                                    )
+                    else:
+                        for dex, xs in enumerate(self.catch[key].xs_entry):
+                            dex = self.catch[key].index[dex]
+                            self.xs_lib[L][m][n][xs]\
+                                .extend([self.get_values(k, dex)])
+
+    def get_values(self, k, index):
+        val = list(np.array(self.lines[k+1].split()).astype(float))
+        return val[index]
+
+    def get_multi_line_values(self, k):
+        values = []
+        while True:
+            val = self.lines[k+1].split()
+            k += 1
+            for ent in val:
+                try:
+                    values.extend([float(ent)])
```

Comment:
use values.append(float(ent)) rather than extend. extend makes sense if you want to add on a list, but adding a list of length one should be done using append.